primefaces/src/main/resources/META-INF/resources/primefaces/inputmask/1-inputmask.js

```diff
@@ -33,6 +33,11 @@ PrimeFaces.widget.InputMask = PrimeFaces.widget.BaseWidget.extend({
      */
     applyMask: function() {
         if(this.cfg.mask && !this.jq.is('[readonly]') && !this.jq.is(':disabled')) {
+            var autoClear = (this.cfg.autoClear === undefined) ? true : this.cfg.autoClear;
+            if (autoClear) {
+                this.cfg.clearMaskOnLostFocus = true;
```

Comment:
Shouldn't this `this.cfg.clearMaskOnLostFocus = true;` be set regardless of autoClear value for it to work when autoClear=false

Reply:
Good catch. Fixed!   I will only set the other one if autoClear = true because its false by default